.github/workflows/build.yml

```diff
@@ -1,85 +1,93 @@
-on: [push, pull_request]
-
 name: build
 
+on:
+  push:
+    branches: [master]
+  pull_request:
+
 jobs:
   build:
-    name: Lints and Test
+    name: Build
     strategy:
+      fail-fast: false
       matrix:
         platform: [ubuntu-latest, macos-latest, windows-latest]
         toolchain: [stable]
     runs-on: ${{ matrix.platform }}
 
     steps:
-      - name: Checkout sources
+      - name: Checkout Sources
         uses: actions/checkout@v2
 
-      - name: Install Rust toolchain
+      - name: Cache Dependencies & Build Outputs
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Install Rust Toolchain
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: ${{ matrix.toolchain }}
           override: true
           components: rustfmt, clippy
 
-      - name: Run cargo fmt
+      - name: Check Code Format
         uses: actions-rs/cargo@v1
         with:
           command: fmt
           args: --all -- --check
 
-      - name: Run cargo clippy
+      - name: Code Lint
         uses: actions-rs/cargo@v1
         with:
           command: clippy
           args: --all-targets --all-features --workspace -- -D warnings
 
-      - name: Run cargo test
+      - name: Test
         uses: actions-rs/cargo@v1
         with:
           command: test
           args: --all-features --workspace
 
-  coverage:
-    name: Code Coverage
+  ensure_no_std:
+    name: Ensure no_std
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout sources
-        uses: actions/checkout@v2
-
-      - name: Install stable Rust toolchain
-        uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: stable
           override: true
-
-      - name: Run cargo-tarpaulin
-        uses: actions-rs/tarpaulin@v0.1
+          target: wasm32-unknown-unknown
+      - name: Build
+        uses: actions-rs/cargo@v1
         with:
-          args: '--all-features -- --test-threads 1'
-
-      - name: Upload to codecov.io
-        uses: codecov/codecov-action@v1.0.6
+          command: build
+          args: --no-default-features --target wasm32-unknown-unknown
```

Comment:
This one fails for me locally, due to `core` not being there.

Based on https://blog.dbrgn.ch/2019/12/24/testing-for-no-std-compatibility/ `thumbv6m-none-eabi` should work for ensuring `no_std` compatibility.

Reply:
The code from https://github.com/multiformats/rust-multibase/blob/c87cf2bafbd1e808ed75bb9e7ebf6d80b023fd33/.github/workflows/build.yml#L43-L56 can be used.